pkg/controller/deployment/sync.go

```diff
@@ -510,15 +510,26 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
 	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
 	if err != nil {
-		return false
+		return false, err
 	}
 	// If there is no new replica set matching this deployment and the deployment isn't paused
 	// then there is a new rollout that waits to happen
 	if newRS == nil && !d.Spec.Paused {
```

Comment:
@bgrant0607 Would you be able to provide some technical direction here?

@kargakis If we want to support the use case you suggested, I think we need a scaling strategy that allows the user to configure which RS will be scaled.  The use case you suggested is broken by proportional scaling generally.  Supporting it when you scale+update in 1 update, but not when you update then scale in 2 updates does not seem intuitive to me.  We could allow the user to specify options: PROPORTIONAL & LATEST.  PROPORTIONAL would always scale RS based on their % of the total replicas, and LATEST would always scale the newest RS.

Scaling the newest RS sometimes (when we observe that we need to create a new RS and scale during the same control loop) and proportionally scaling all the RS other times (when we observe only that we need to scale during the control loop) seems inconsistent to me and the behavior is non-deterministic from the perspective of the user - most of the time it will do what they expect, but occasionally it may do something different.

What the user really desires will likely vary from case to base, but I think we should be as deterministic as possible with how we perform scaling.

All that said... the behavior of HEAD is broken right now.  This doesn't just impact scaling + rollout in the same update, it will break any time a scale and update are seen in the control loop even if they happened separately.  I want to get this fixed quickly, and the PR as written is at least functional.  If we commit this PR would you be willing to address the concerns in an immediate follow up?